plainly-plugin/src/ui/components/settings/AuthProvider.tsx

```diff
@@ -0,0 +1,44 @@
+import { LoaderCircleIcon } from 'lucide-react';
+import { createContext } from 'react';
+import { useSessionStorage } from '../../hooks/useSessionStorage';
+import { useSettings } from '../../hooks/useSettings';
+import MissingApiKey from './MissingApiKey';
+import PinOverlay from './PinOverlay';
+
+interface AuthContextProps {
+  apiKey: string;
+}
+
+export const AuthContext = createContext<AuthContextProps>(
+  {} as AuthContextProps,
+);
+
+export const AuthProvider = ({ children }: { children: React.ReactNode }) => {
+  const { getSettingsApiKey, loading } = useSettings();
+
+  const [pin, setPinStorage] = useSessionStorage<string | undefined>(
+    'pin',
+    undefined,
+  );
+
+  const { key, encrypted, error } = getSettingsApiKey(pin);
```

Comment:
* ok imo this is wrong definatelly, `apiKeySet` and `apiKeyEcnrypted` should be provided from the `useSettings()`
* the `PinOverlay` onSubmit function should:
   * call the getApiKey();
   * if error -> setError
   * if ok -> update session storage (which will trigger rerender and thus set auto context provider to be sivible)